src/data_management/timeseries/locations/EditLocation.tsx

```diff
@@ -20,16 +23,43 @@ const EditLocation = (props: RouteProps & RouteComponentProps) => {
         credentials: "same-origin"
       }).then(response => response.json());
 
+      if (currentRecord.object && currentRecord.object.type !== null && currentRecord.object.id !== null) {
+        setRelatedAssetRequired(true);
+      } else {
+        setRelatedAssetRequired(false);
+      }
+      setRelatedAsset(null);
       setCurrentRecord(currentRecord);
     })();
   }, [uuid])
 
+  useEffect (() => {
+    
+      (async () => {
+          // Todo: how to fix this error? I already check if currentRecord === null
+          // @ts-ignore
+          if (relatedAssetRequired && currentRecord && currentRecord !== null && currentRecord !== undefined && currentRecord.object) {
+            // Todo: how to fix this error? I already check if currentRecord === null
+            // @ts-ignore
+            const relatedAsset = await fetch(`/api/v3/${currentRecord.object.type}s/${currentRecord.object.id}/`, {
```

Comment:
This is because you are missing a type definition for currentRecord when you define it:

for example: `const [currentRecord, setCurrentRecord] = useState(null)` should become `useState<any>(null)` (use any or make type definition for it). In your case, currentRecord can only be null.

The same is applied to relatedAsset and relatedAssetRequired, there is no type definition for them.